channeldb/invoices.go

```diff
@@ -793,9 +846,22 @@ func (d *DB) LookupInvoice(ref InvoiceRef) (Invoice, error) {
 // reference. The payment address will be treated as the primary key, falling
 // back to the payment hash if nothing is found for the payment address. An
 // error is returned if the invoice is not found.
-func fetchInvoiceNumByRef(invoiceIndex, payAddrIndex kvdb.RBucket,
+func fetchInvoiceNumByRef(invoiceIndex, payAddrIndex, setIDIndex kvdb.RBucket,
 	ref InvoiceRef) ([]byte, error) {
 
+	// If the set id is present, we only consult the set id index for this
```

Comment:
should this rather go in its own method, since in this case only the single argument is used:

```go
var invoiceNum []byte
if setIDIndex != nil {
     invoiceNum, err = fetchInvoiceBySetId(setIdIndex)
else {
     invoiceNum, err = fetchInvoiceNumByRef(invoiceIndex, payAddrIndex, ref)
}
```

Reply:
Same could be said for the usages of `payAddrIndex` below. IIUC, the idea here is to have the `fetchInvoiceNUmByRef` hide some of the details re the various indexes we now have in the database.

Reply:
it could, i think there was some desire to have everything funneled through one place so that the access is always consistent though. we can definitely look into splitting it up if it gets unwieldy